datafusion/optimizer/src/optimizer.rs

```diff
@@ -506,16 +532,16 @@ mod tests {
             produce_one_row: false,
             schema: Arc::new(DFSchema::empty()),
         });
-        let err = opt.optimize(&plan, &config, &observe).unwrap_err();
+        let err = opt.optimize(plan, &config, &observe).unwrap_err();
         assert_eq!(
             "Optimizer rule 'get table_scan rule' failed\ncaused by\nget table_scan rule\ncaused by\n\
             Internal error: Failed due to a difference in schemas, \
-            original schema: DFSchema { fields: [\
```

Comment:
this test actually shows the output was reversed -- the original schema was `[]` not the table schema (which is what is changed by the optimizer)